app/controllers/v1/supported_ssgs_controller.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+require 'yaml'
+
+module V1
+  # API for Benchmarks
+  class SupportedSsgsController < ApplicationController
+    # Pseudo-resource
+    SupportedSsg = Struct.new(
+      :id, :package, :version, :profiles,
+      :os_major_version, :os_minor_version,
+      keyword_init: true
+    )
+
+    def index
+      render_json scope_search
+    end
+
+    private
+
+    def metadata(opts = {})
+      opts[:total] ||= scope_search.count
+      {
+        meta: {
+          total: opts[:total],
+          revision: raw_supported['revision']
+        }
+      }
+    end
+
+    def serializer
+      SupportedSsgSerializer
+    end
+
+    def supported
+      raw_supported['supported'].map do |rhel, values|
+        major, minor = os_version(rhel)
+
+        SupportedSsg.new(
+          id: rhel,
+          os_major_version: major,
+          os_minor_version: minor,
+          **map_attributes(values)
+        )
+      end
+    end
+    alias scope_search supported
+
+    def map_attributes(values)
+      attrs = values.slice(*SupportedSsg.members.map(&:to_s))
+      attrs['profiles'] = attrs['profiles']&.keys
+      attrs.symbolize_keys
+    end
+
+    def os_version(rhel)
+      rhel.scan(/(\d+)\.(\d+)$/)[0]
+    end
+
+    def raw_supported
+      @raw_supported ||= YAML.load_file(supported_file)
+    end
+
+    def supported_file
+      Rails.root.join('config/supported_ssg.yaml')
+    end
```

Comment:
Looking great, @vkrizan! One initial thing I see here: I think it'd be good to write a pseudo model or service that reads this config and defines the struct, rather than putting it all in this controller.

Reply:
Any advise where to introduce it? In services or models (there are only db ones)?

Reply:
It's fine to put it with the models IMO. The only thing to leave off is the inheritance from `ApplicationRecord`.

Reply:
Done, and ready for review.